src/cargo/util/to_url.rs

```diff
@@ -40,3 +60,38 @@ fn mapper(s: &str) -> url::SchemeType {
         s => url::whatwg_scheme_type_mapper(s),
     }
 }
+
+fn determine_base_url() -> Result<Url, String> {
+    let current_dir = try!(env::current_dir().map_err(|s| {
```

Comment:
There are roughly 3 options, I think:

1. Relative to the current working directory (as is). Shouldn't be too surprising to people, but I think it would mean that `cargo build` would now only work if you're in the project root. I think this is a regression.
2. Relative to the project root.
3. Relative to the file that defines the URL (your suggestion).

I'm most in favor of option 2. However, both 2/3 would require plumbing. In option 3, all calls to `to_url` would require an additional argument. In option 2, we could use `CARGO_MANIFEST_DIR` to avoid the additional argument.